src/writer.rs

```diff
@@ -27,12 +27,13 @@ enum HeaderReference<'a> {
 impl<'a> HeaderReference<'a> {
     fn to_header(&self) -> Result<NiftiHeader> {
         match self {
-            HeaderReference::FromHeader(h) => Ok(h.clone().to_owned()),
+            HeaderReference::FromHeader(h) => Ok((*h).to_owned()),
```

Comment:
I would have thought that `h.to_owned()` would work. Is it because `h` is a `&&NiftiHeader` (according to rust-analyzer)?

Reply:
Indeed, it is a reference to another reference, which makes `(*h)` necessary. Deref rules are sometimes a bit awkward. :)